scripts/pm/hf_from_gcs.py

```diff
@@ -0,0 +1,224 @@
+                step_dirs_local.append((full_path, step_number))
+                print(f"Found step directory: {full_path} with step {step_number}")
+            except (IndexError, ValueError) as e:
+                print(f"Error parsing step number from {dir_name}: {e}")
+
+    print(f"Found {len(step_dirs_local)} step directories in {gcs_path}")
+    return step_dirs_local
+
+
+def download_from_gcs(gcs_path, local_path):
+    """Download contents from a GCS path to a local directory using gsutil."""
+    try:
+        print(f"Downloading {gcs_path} to {local_path}...")
+        cmd = ["gsutil", "-m", "cp", "-r", f"{gcs_path}*", local_path]
+        result = subprocess.run(cmd, check=True, capture_output=True, text=True)
+        print(f"Download completed successfully.")
+        return True
+    except subprocess.CalledProcessError as e:
+        print(f"Error downloading from {gcs_path}: {e}")
+        print(f"stdout: {e.stdout}")
+        print(f"stderr: {e.stderr}")
+        return False
+
+
+def upload_to_huggingface(local_path, repo_id, revision):
+    """Upload a local directory to Hugging Face as a specific revision."""
+    try:
+        print(f"Uploading checkpoint to Hugging Face as revision: {revision}")
+        # Make sure huggingface_hub is installed
+        subprocess.run(["pip", "install", "huggingface_hub"], check=True, capture_output=True)
+
+        # Use the Hugging Face CLI to upload
+        cmd = [
+            "python",
+            "-m",
+            "huggingface_hub",
+            "upload",
+            local_path,
+            "--repo-id",
+            repo_id,
+            "--revision",
+            f"step-{revision}",
+            "--commit-message",
+            f"Upload checkpoint for step {revision}",
+        ]
+        result = subprocess.run(cmd, check=True, capture_output=True, text=True)
+        print(f"Upload completed successfully.")
+        print(result.stdout)
+        return True
+    except subprocess.CalledProcessError as e:
+        print(f"Error uploading to Hugging Face: {e}")
+        print(f"stdout: {e.stdout}")
+        print(f"stderr: {e.stderr}")
+        return False
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Upload checkpoints from GCS to Hugging Face")
+    parser.add_argument(
+        "--repo-id", required=True, help='Target Hugging Face repository ID (e.g., "username/model-name")'
+    )
+    parser.add_argument("--dry-run", action="store_true", help="Only list checkpoints without uploading")
+    args = parser.parse_args()
+
+    # Initialize the GCS client
+    client = storage.Client()
+
+    # Collect all step directories
+    all_step_dirs = []
+
+    # Main execution
+    for directory in dirs:
+        try:
+            step_dirs = list_gcs_directories(directory)
+            all_step_dirs.extend(step_dirs)
+        except Exception as e:
+            print(f"Error listing {directory}: {e}")
+
+    # Sort all step directories by step number
```

Comment:
Why do they have to be sorted by step number?

Reply:
These will all get uploaded as commits to a single repo for the Marin 8B model, so we want to push those commits in chronological order.